src/components/ChatBox.tsx

```diff
@@ -0,0 +1,281 @@
+        if (data.type === "SEND") {
+          if (
+            (data.content.senderEmail === store.user.profile?.email &&
+              data.content.seenByRecipient === true) ||
+            data.content.senderEmail !== store.user.profile?.email
+          ) {
+            if (data.content.seenByRecipient === true) {
+              setLastSeenMessage(data.content.id);
+            }
+          }
+          setMessages((prevMessages) => [...prevMessages, data.content]);
+        } else {
+          if (data.type === "JOIN") {
+            handleJoin(data.joiningUserEmail);
+          }
+        }
+      },
+      headers
+    );
+  };
+
+  const handleJoin = (joiningUserEmail: string | undefined) => {
+    if (joiningUserEmail !== store.user.profile?.email) {
+      //recipient joined = they read all messages
+      if (messagesRef.current.length > 0) {
+        setLastSeenMessage(
+          messagesRef.current[messagesRef.current.length - 1].id
+        );
+      }
+    } else {
+      if (doesChatRoomExist === true && chatId !== null) {
+        getMessages();
+      }
+    }
+  };
+
+  const getMessages = async () => {
+    const messagesResponse = await api.getMessagesFromSpecifiedChatRoom(
+      chatId!,
+      "0",
+      "10",
+      timeZone
+    );
+    setMessages([...messagesResponse.content].reverse());
+
+    if (messagesResponse.content.length > 0) {
+      if (
+        messagesResponse.content[0].senderEmail !== store.user.profile?.email
+      ) {
+        // if last message was sent by recipient we mark it as lastSeenMessage
+        setLastSeenMessage(messagesResponse.content[0].id);
+      } else {
+        setLastSeenMessage(
+          messagesResponse.content.find(
+            (message) => message.seenByRecipient === true
+          )?.id
+        );
+      }
+    }
+  };
+
+  //TODO: timezone
+  const initializeChatRoom = async (message: string) => {
+    console.log("initializing chat room");
+    const data = await api.initializeChatRoom(
+      recipientEmail,
+      message,
+      timeZone
+    );
+    setDoesChatRoomExist(true);
+    setChatId(data.chatId);
+    setMessages((prevMessages) => [
+      ...prevMessages,
+      {
+        id: 1,
+        chatId: data.chatId,
+        senderEmail: store.user.profile!.email!,
+        content: message,
+        createdAt: new Date().toISOString(),
+        seenByRecipient: false,
+      },
+    ]);
+  };
+
+  const sendMessageToChatRoom = (message: string) => {
+    const headers = {
+      "Accept-Role":
+        store.user.profile!.selected_profile?.toUpperCase() as string,
+      "Accept-Timezone": timeZone,
+    };
+    if (wsClient && message) {
+      const chatMessage = { content: message };
+      wsClient.publish({
+        destination: `/app/chat/${chatId}`,
+        body: JSON.stringify(chatMessage),
+        headers: headers,
+      });
+    } else console.log("sending message ended without sending message");
```

Comment:
Na koniec, usunalbym tego console.loga
```suggestion
    }
```